src/Analyzers/MSTest.Analyzers/UseDeploymentItemWithTestMethodOrTestClassAnalyzer.cs

```diff
@@ -0,0 +1,97 @@
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestMethodAttribute, out INamedTypeSymbol? testMethodAttributeSymbol)
+                 && context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestClassAttribute, out INamedTypeSymbol? testClassAttributeSymbol)
+                 && context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingDeploymentItemAttribute, out INamedTypeSymbol? deploymentItemAttributeSymbol))
+            {
+                context.RegisterSymbolAction(
+                    context => AnalyzeSymbol(context, testMethodAttributeSymbol, testClassAttributeSymbol, deploymentItemAttributeSymbol),
+                    new[] { SymbolKind.NamedType, SymbolKind.Method });
+            }
+        });
+    }
+
+    private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol testMethodAttributeSymbol, INamedTypeSymbol testClassAttributeSymbol, INamedTypeSymbol deploymentItemAttributeSymbol)
+    {
+        bool hasDeploymentItemAttribute = false;
+        bool isTestMethodOrTestClass = false;
+        if (context.Symbol is INamedTypeSymbol namedTypeSymbol && namedTypeSymbol.TypeKind == TypeKind.Class)
+        {
+            foreach (AttributeData attribute in namedTypeSymbol.GetAttributes())
+            {
+                if (attribute.AttributeClass.Inherits(testClassAttributeSymbol))
+                {
+                    isTestMethodOrTestClass = true;
+                }
+                else if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, deploymentItemAttributeSymbol))
+                {
+                    hasDeploymentItemAttribute = true;
+                }
+            }
+
+            if (hasDeploymentItemAttribute && !isTestMethodOrTestClass)
+            {
+                context.ReportDiagnostic(namedTypeSymbol.CreateDiagnostic(UseDeploymentItemWithTestMethodOrTestClassRule));
+            }
+        }
+        else if (context.Symbol is IMethodSymbol methodSymbol)
+        {
+            foreach (AttributeData attribute in methodSymbol.GetAttributes())
+            {
+                if (attribute.AttributeClass.Inherits(testMethodAttributeSymbol))
+                {
+                    isTestMethodOrTestClass = true;
+                }
+                else if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, deploymentItemAttributeSymbol))
+                {
+                    hasDeploymentItemAttribute = true;
+                }
+            }
+
+            if (hasDeploymentItemAttribute && !isTestMethodOrTestClass)
+            {
+                context.ReportDiagnostic(methodSymbol.CreateDiagnostic(UseDeploymentItemWithTestMethodOrTestClassRule));
+            }
+        }
```

Comment:
Please review in code, I might have done some typo as I am doing with GH UI

```suggestion
        bool hasDeploymentItemAttribute = false;
        bool isTestMethodOrTestClass = false;
        foreach (AttributeData attribute in context.Symbol.GetAttributes())
		{
			if (context.Symbol.Kind == SymbolKind.NamedType && attribute.AttributeClass.Inherits(testClassAttributeSymbol))
			{
				isTestMethodOrTestClass = true;
			}
			else if (context.Symbol.Kind == SymbolKind.Method && attribute.AttributeClass.Inherits(testMethodAttributeSymbol))
			{
				isTestMethodOrTestClass = true;
			}
			else if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, deploymentItemAttributeSymbol))
			{
				hasDeploymentItemAttribute = true;
			}
		}
		
		if (hasDeploymentItemAttribute && !isTestMethodOrTestClass)
		{               
			context.ReportDiagnostic(methodSymbol.CreateDiagnostic(UseDeploymentItemWithTestMethodOrTestClassRule));
		}
```